blocks/components/editable/index.js

```diff
@@ -319,6 +388,27 @@ export default class Editable extends wp.element.Component {
 		}
 	}
 
+	isEmpty( value ) {
+		// Empty array.
+		if ( ! value.length ) {
+			return true;
+		}
+
+		const props = value[ 0 ].props;
```

Comment:
I'm having a hard time following the cases we're checking here. To me it almost seems like it should only ever be considered empty if `value` is an empty array. Why do we have empty strings or empty elements in `value` in the first place? Can we avoid that?

Otherwise, perhaps an `Array#every` is in order:

```js
return value.every( ( child ) => {
	if ( 'string' === typeof child ) {
		return child.length > 0;
	}

	// Recurse to check emptiness of children
	return ! this.isEmpty( child.props.children );
} );
```